libs/game/game.ts

```diff
@@ -112,7 +118,48 @@ namespace game {
         __background.viewY += dy;
     }
 
-    function showBackground(h: number, c: number) {
+    /**
+     * Sets the map for rendering tiles
+     * @param map 
+     */
+    //% blockId=gamesettilemap block="set tile map to %map"
+    //% map.fieldEditor="sprite"
+    //% map.fieldOptions.taggedTemplate="img"
+    //% group="Tiles"
+    export function setTileMap(map: Image) {
+        if (!__tileMap)
+            __tileMap = new tiles.TileMap(16, 16);
+        __tileMap.setMap(map);
+    }
+
+    /**
+     * Sets the tile image at the given index
+     * @param index 
+     * @param img 
+     */
+    //% img.fieldEditor="sprite"
+    //% img.fieldOptions.taggedTemplate="img"
+    //% blockId=gamesettile block="set tile at %index to %img||with collisions %collisions=toggleOnOff"
+    //% group="Tiles"
+    export function setTile(index: number, img: Image, collisions?: boolean) {
+        if (!__tileMap)
+            __tileMap = new tiles.TileMap(img.width, img.height);
+        __tileMap.setTile(index, img, !!collisions);
```

Comment:
Casting Boolean to a Boolean? If you're checking for null/undefined you can give collisions a default value of false instead.